File: Data Structures & Algorithms/trapping-rain-water/submission-2.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        int l=0;
        int r=height.size()-1;
        int left=height[l];
        int right=height[r];
        int res=0;
        while(l<r){
            if(left<right){
                l++;
                left=max(left,height[l]);
                res+=left-height[l];
            }else{
                r--;
                right=max(right,height[r]);
                res+=right-height[r];
            }
        }
        return res;
        
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isBalanced(TreeNode root) {
        int res[]=new int[1];
        check(root,res);
        if(res[0]==1){
            return false;
        }else{
            return true;
        }
        
        
    }
    private int check(TreeNode node,int[] res){
        if(node==null) return 0;
        int left=check(node.left,res);
        int right=check(node.right,res);
        if(Math.abs(right-left)>1){
            res[0]=1;
        }
        return 1+Math.max(left,right);
    }
}
